openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java

```diff
@@ -0,0 +1,250 @@
+
+  public LdapAuthenticator(
+      UserRepository userRepository, TokenRepository tokenRepository, OpenMetadataApplicationConfig config) {
+    if (config.getAuthenticationConfiguration().getProvider().equals("ldap")
+        && config.getAuthenticationConfiguration().getLdapConfiguration() != null) {
+      ldapLookupConnectionPool = getLdapConnectionPool(config.getAuthenticationConfiguration().getLdapConfiguration());
+    } else {
+      throw new RuntimeException("Invalid or Missing Ldap Configuration.");
+    }
+    this.userRepository = userRepository;
+    this.tokenRepository = tokenRepository;
+    this.ldapConfiguration = config.getAuthenticationConfiguration().getLdapConfiguration();
+    this.loginAttemptCache = new LoginAttemptCache(config);
+  }
+
+  private LDAPConnectionPool getLdapConnectionPool(LdapConfiguration ldapConfiguration) {
+    try {
+      if (ldapConfiguration.getSslEnabled()) {
+        AggregateTrustManager trustManager =
+            new AggregateTrustManager(
+                false,
+                JVMDefaultTrustManager.getInstance(),
+                new TrustStoreTrustManager(
+                    ldapConfiguration.getKeyStorePath(),
+                    ldapConfiguration.getKeyStorePassword().toCharArray(),
+                    ldapConfiguration.getTruststoreFormat(),
+                    true));
+        SSLUtil sslUtil = new SSLUtil(trustManager);
+
+        LDAPConnectionOptions connectionOptions = new LDAPConnectionOptions();
+        SSLSocketVerifier sslSocketVerifier =
+            ldapConfiguration.getVerifyCertificateHostname()
+                ? new HostNameSSLSocketVerifier(true)
+                : TrustAllSSLSocketVerifier.getInstance();
+        connectionOptions.setSSLSocketVerifier(sslSocketVerifier);
+
+        try (LDAPConnection connection =
+            new LDAPConnection(
+                sslUtil.createSSLSocketFactory(),
+                connectionOptions,
+                ldapConfiguration.getHost(),
+                ldapConfiguration.getPort())) {
+          // Use the connection here.
+          return new LDAPConnectionPool(connection, ldapConfiguration.getMaxPoolSize());
+        } catch (GeneralSecurityException e) {
+          LOG.warn("[LDAP] Issue in creating a LookUp Connection SSL");
+        }
+      } else {
+        LDAPConnection conn =
+            new LDAPConnection(
+                ldapConfiguration.getHost(),
+                ldapConfiguration.getPort(),
+                ldapConfiguration.getDnAdminPrincipal(),
+                ldapConfiguration.getDnAdminPassword());
+        return new LDAPConnectionPool(conn, ldapConfiguration.getMaxPoolSize());
+      }
+    } catch (LDAPException e) {
+      LOG.warn("[LDAP] Issue in creating a LookUp Connection");
+    }
+    return null;
+  }
+
+  @Override
+  public JwtResponse loginUser(LoginRequest loginRequest) throws IOException, TemplateException {
+    // validate Login is Not Blocked
```

Comment:
same here, lets remove the comments. Can this be the default implementation as it look the same to BasicAuth